timelock-impl/src/main/java/com/palantir/atlasdb/timelock/ConjureTimelockResource.java

```diff
@@ -145,11 +150,19 @@ public ListenableFuture<LeaderTime> leaderTime(AuthHeader authHeader, String nam
     public ListenableFuture<ConjureLockResponse> lock(
             AuthHeader authHeader, String namespace, ConjureLockRequest request) {
         return handleExceptions(() -> {
+            List<LockDescriptor> lockDescriptors = fromConjureLockDescriptors(request.getLockDescriptors());
+            Optional<LockRequestMetadata> metadata = request.getMetadata()
+                    .map(conjureMetadata -> ConjureLockRequestMetadataUtils.fromConjureIndexEncoded(
+                            ConjureMetadataConversionResult.builder()
+                                    .conjureMetadata(conjureMetadata)
+                                    .lockList(lockDescriptors)
+                                    .build()));
             IdentifiedLockRequest lockRequest = ImmutableIdentifiedLockRequest.builder()
-                    .lockDescriptors(fromConjureLockDescriptors(request.getLockDescriptors()))
+                    .lockDescriptors(lockDescriptors)
```

Comment:
Even though we are passing a list to field that expects a set, immutable types always build an immutable version of the container type anyways (to guarantee immutability)